File: dart/basics/word_frequency.dart
// Count the words of a fixed text, most frequent first and alphabetically within a tie.
// Run: dart run word_frequency.dart

const text = 'the quick brown fox jumps over the lazy dog the fox barks';

void main() {
  final counts = <String, int>{};
  for (final word in text.split(RegExp(r'\s+'))) {
    counts.update(word, (value) => value + 1, ifAbsent: () => 1);
  }

  final ranked = counts.entries.toList()
    ..sort((a, b) {
      final byCount = b.value.compareTo(a.value);
      return byCount != 0 ? byCount : a.key.compareTo(b.key);
    });

  for (final entry in ranked) {
    print('${entry.key} ${entry.value}');
  }
}

File: js/basics/word_frequency.js
// Count the words of a fixed text, most frequent first and alphabetically within a tie.
// Run: node word_frequency.js

const text = "the quick brown fox jumps over the lazy dog the fox barks";

const counts = new Map();
for (const word of text.split(/\s+/)) {
  counts.set(word, (counts.get(word) ?? 0) + 1);
}

const ranked = [...counts].sort(
  (a, b) => b[1] - a[1] || (a[0] < b[0] ? -1 : a[0] > b[0] ? 1 : 0),
);
for (const [word, count] of ranked) {
  console.log(`${word} ${count}`);
}
